tests/core/test_utils.py

```diff
@@ -271,14 +277,47 @@ def _get_all_aws_regions():
     return regions
 
 
+@pytest.fixture()
+def test_arn():
+    arn = "arn:aws:sagemaker:{region}:012345678910:processing-job/random-test-arn"
+    return arn
+
+
+def false_get(*args, **kwargs):
+    assert False, "GET Should Not Be Called"
+
+
+def true_get(*args, **kwargs):
+    print("GET Should Be Called")
+    assert True
+
+
 @pytest.mark.parametrize("region", _get_all_aws_regions())
-def test_telemetry_url_preparation(region):
-    test_arn = "arn:aws:sagemaker:{region}:012345678910:processing-job/random-test-arn"
+def test_telemetry_url_preparation(test_arn, region):
     arn_with_region = test_arn.format(region=region)
     assert get_aws_region_from_processing_job_arn(arn_with_region) == region
-    url = prepare_telemetry_url(arn_with_region)
+    url = _prepare_telemetry_url(arn_with_region)
     assert url == PROFILER_TELEMETRY_URL.format(
         region=region
     ) + "/?x-artifact-id={report_version}&x-arn={arn}".format(
         report_version=PROFILER_REPORT_VERSION, arn=arn_with_region
     )
+
+
+def test_setup_profiler_report(monkeypatch, test_arn):
+    test_arn = test_arn.format(region="us-east-1")
+    monkeypatch.setattr(requests, "get", true_get)
+
+    # setup_profiler_report is expected to be executed when called with a correct ARN
+    setup_profiler_report(test_arn)
+
+    # setup_profiler_report is expected to be executed when called with opt_out=False
+    setup_profiler_report(test_arn, opt_out=False)
```

Comment:
How are you verifying that `setup_profiler_report` was called? Not sure if this is possible with monkeypatch, but using the `@patch` decorator, you can check the number of calls made to a mocked function. So one idea is to "mock" the `requests.get` function as you've already done so you can simply check the number of times it was called.

[Example](https://stackoverflow.com/a/21983089)

Reply:
monkeypatch does exactly that. If `false_get` is executed, the test fails